tools/expotools/src/commands/CheckPackages.ts

```diff
@@ -1,151 +1,61 @@
+    .option('--no-build', 'Whether to skip `yarn build` check.', false)
+    .option('--no-test', 'Whether to skip `yarn test` check.', false)
+    .option('--no-lint', 'Whether to skip `yarn lint` check.', false)
+    .option('--fix-lint', 'Whether to run `yarn lint --fix` instead of `yarn lint`.', false)
+    .option(
+      '--no-uniformity-check',
+      'Whether to check the uniformity of committed and generated build files.',
+      false
+    )
+    .description('Checks if packages build successfully and their tests pass.')
+    .asyncAction(main);
+};
 
+async function main(packageNames: string[], options: ActionOptions): Promise<void> {
+  options.packageNames = packageNames;
+
+  const packages = await getPackagesToCheckAsync(options);
+  const failedPackages: string[] = [];
   let passCount = 0;
 
   for (const pkg of packages) {
-    if (!pkg.scripts.build && !pkg.scripts.test) {
-      // If the package doesn't have build or test script, just skip it.
-      continue;
-    }
-    if (only.length > 0 && !only.includes(pkg.packageName)) {
-      continue;
-    }
-    console.log(`🔍 Checking the ${chalk.bold.green(pkg.packageName)} package ...`);
-
-    try {
-      if (options.build) {
-        await runScriptAsync(pkg, 'clean');
-        await runScriptAsync(pkg, 'build');
-
-        if (options.uniformityCheck) {
-          await checkBuildUniformityAsync(pkg);
-        }
-      }
-      if (options.test) {
-        const args = ['--watch', 'false', '--passWithNoTests'];
-
-        if (process.env.CI) {
-          // Limit to one worker on CIs
-          args.push('--maxWorkers', '1');
-        }
-        await runScriptAsync(pkg, 'test', args);
-      }
-      if (options.lint) {
-        const args = ['--max-warnings', '0'];
-        if (options.fixLint) {
-          args.push('--fix');
-        }
-        await runScriptAsync(pkg, 'lint', args);
-      }
-      console.log(`✨ ${chalk.bold.green(pkg.packageName)} checks passed.`);
+    if (await checkPackageAsync(pkg, options)) {
       passCount++;
-    } catch (error) {
+    } else {
       failedPackages.push(pkg.packageName);
     }
-    console.log();
+    logger.log();
   }
 
   const failureCount = failedPackages.length;
 
-  if (failureCount === 0) {
-    console.log(chalk.bold.green(`🏁 All ${passCount} packages passed.`));
-    process.exit(0);
-  } else {
-    console.log(
-      `${chalk.green(`🏁 ${passCount} packages passed`)},`,
-      `${chalk.magenta(`${failureCount} ${failureCount === 1 ? 'package' : 'packages'} failed:`)}`,
-      failedPackages.map((failedPackage) => chalk.yellow(failedPackage)).join(', ')
+  if (failureCount !== 0) {
+    logger.log(
+      `${green(`🏁 ${passCount} packages passed`)},`,
+      `${magenta(`${failureCount} ${failureCount === 1 ? 'package' : 'packages'} failed:`)}`,
+      failedPackages.map((failedPackage) => yellow(failedPackage)).join(', ')
     );
-    process.exit(1);
+    return process.exit(1);
```

Comment:
Put `return` in its own statement in functions that return void
```suggestion
    process.exit(1);
    return;
```